internal/controller/configmap.go

```diff
@@ -110,37 +113,158 @@ func newConfigMap() *corev1.ConfigMap {
 }
 
 // reconcileRBAC will ensure that the ArgoCD RBAC ConfigMap is present.
-func (r *RoleReconciler) reconcileRBAC(role *argoprojiov1alpha1.Role) error {
+func (r *ArgoCDRoleReconciler) reconcileRBAC(role *argoprojiov1alpha1.ArgoCDRole) error {
 	cm := newConfigMap()
 	if IsObjectFound(r.Client, cm.Namespace, cm.Name, cm) {
 		return r.reconcileRBACConfigMap(cm, role)
 	}
 	return r.createRBACConfigMap(cm, role)
 }
 
+// reconcileRBACWithRoleBinding will ensure that the ArgoCD RBAC ConfigMap is present.
+func (r *ArgoCDRoleReconciler) reconcileRBACWithRoleBinding(role *argoprojiov1alpha1.ArgoCDRole, rb *argoprojiov1alpha1.ArgoCDRoleBinding) error {
+	cm := newConfigMap()
+	if IsObjectFound(r.Client, cm.Namespace, cm.Name, cm) {
+		return r.reconcileRBACConfigMapWithRoleBinding(cm, role, rb)
+	}
+	return r.createRBACConfigMapWithRoleBinding(cm, role, rb)
+}
+
 // reconcileRBACConfigMap will ensure that the ArgoCD RBAC ConfigMap is up-to-date.
-func (r *RoleReconciler) reconcileRBACConfigMap(cm *corev1.ConfigMap, role *argoprojiov1alpha1.Role) error {
+func (r *ArgoCDRoleReconciler) reconcileRBACConfigMap(cm *corev1.ConfigMap, role *argoprojiov1alpha1.ArgoCDRole) error {
+	changed := false
+	overlayKey := fmt.Sprintf("policy.%s.csv", role.Name)
```

Comment:
same as in another comment: roles from multiple namespaces could have the same name and conflict in their policy key

Reply:
@sbeginCoveo yea we either make a new overlayKey or don't allow role creation outside of a specific namespace

Reply:
`"policy.${namespace}.${rolename}.csv"` could be a pretty simple fix for this :)